test/controllers/provisioning_templates_controller_test.rb

```diff
@@ -132,6 +132,12 @@ class ProvisioningTemplatesControllerTest < ActionController::TestCase
       assert_redirected_to provisioning_templates_path
     end
 
+    test "build menu should return with error code if no TFTP defined" do
+      SmartProxy.stubs(:with_features).with('TFTP').returns('')
+      get :build_pxe_default, {}, set_session_user
+      assert_response 422
```

Comment:
This action always redirects, it doesn't return a 4xx status code to the browser.